src/operator/operator.go

```diff
@@ -0,0 +1,746 @@
+	Eq(a any) (any, error)
+}
+
+type INe interface {
+	Ne(a any) (any, error)
+}
+
+type ILt interface {
+	Lt(a any) (any, error)
+}
+
+type ILe interface {
+	Le(a any) (any, error)
+}
+
+type IGt interface {
+	Gt(a any) (any, error)
+}
+
+type IGe interface {
+	Ge(a any) (any, error)
+}
+
+type IBool interface {
+	Bool() (bool, error)
+}
+
+type IPos interface {
+	Pos() (any, error)
+}
+
+type INeg interface {
+	Neg() (any, error)
+}
+
+type IContains interface {
+	Contains(a any) (bool, error)
+}
+
+func Mul(a any, b any) (any, error) {
+	if imul, ok := a.(IMul); ok {
+		return imul.Mul(b)
+	}
+	if irmul, ok := b.(IRMul); ok {
+		return irmul.RMul(b)
+	}
+	if IsNumeric(a) {
+		if IsNumeric(b) {
+			return multiplyNumeric(a, b), nil
+		}
+		if aI, ok := ToInt(a); ok {
+			switch reflect.TypeOf(b).Kind() {
+			case reflect.Slice, reflect.Array:
+				return mulSliceByInt(b, aI), nil
+			case reflect.String:
+				return mulStrByInt(b.(string), aI), nil
+			}
+		}
+	} else {
+		if bI, ok := ToInt(b); ok {
+			switch reflect.TypeOf(a).Kind() {
+			case reflect.Slice, reflect.Array:
+				return mulSliceByInt(a, bI), nil
+			case reflect.String:
+				return mulStrByInt(a.(string), bI), nil
+			}
+		}
+	}
+	return nil, fmt.Errorf("given elements are not multipliable")
+}
+
+func Add(a any, b any) (any, error) {
+	if iadd, ok := a.(IAdd); ok {
+		return iadd.Add(b)
+	}
+	if irAdd, ok := b.(IRAdd); ok {
+		return irAdd.RAdd(b)
+	}
+	if IsNumeric(a) && IsNumeric(b) {
+		return addNumeric(a, b), nil
+
+	}
+	if bothString(a, b) {
+		return a.(string) + b.(string), nil
+	}
+	if slices.Contains([]reflect.Kind{reflect.Slice, reflect.Array}, reflect.TypeOf(a).Kind()) &&
+		slices.Contains([]reflect.Kind{reflect.Slice, reflect.Array}, reflect.TypeOf(b).Kind()) {
+		return addSlices(a, b), nil
+	}
+
+	return nil, fmt.Errorf("given elements are not additible")
+}
+
+func Sub(a any, b any) (any, error) {
+	if i, ok := a.(ISub); ok {
+		return i.Sub(b)
+	}
+	if ir, ok := b.(IRSub); ok {
+		return ir.RSub(b)
+	}
+	if IsNumeric(a) && IsNumeric(b) {
+		return subNumeric(a, b), nil
+	}
+
+	return nil, fmt.Errorf("given elements are not subable")
+}
+
+func Div(a any, b any) (any, error) {
+	if i, ok := a.(IDiv); ok {
+		return i.Div(b)
+	}
+	if ir, ok := b.(IRDiv); ok {
+		return ir.RDiv(b)
+	}
+	if IsNumeric(a) && IsNumeric(b) {
+		return divNumeric(a, b)
+	}
+
+	return nil, fmt.Errorf("given elements are not divable")
+}
+
+func Mod(a any, b any) (any, error) {
+	if i, ok := a.(IMod); ok {
+		return i.Mod(b)
+	}
+	if i, ok := b.(IRMod); ok {
+		return i.RMod(a)
+	}
+	if IsNumeric(a) && IsNumeric(b) {
+		return modNumeric(a, b)
+	}
+
+	return nil, fmt.Errorf("given elements are not divable")
+}
+
+func Pow(a any, b any) (any, error) {
+	if i, ok := a.(IPow); ok {
+		return i.Pow(b)
+	}
+	if ir, ok := b.(IRPow); ok {
+		return ir.RPow(b)
+	}
+	if IsNumeric(a) && IsNumeric(b) {
+		return powNumeric(a, b), nil
+	}
+
+	return nil, fmt.Errorf("given elements are not powable")
```

Comment:
honesty, who wrote these comments? xD

Reply:
me myself and i

Reply:
nto sure about the mental process but happened but i figured let's do it like name of function + able everywhere xD